### ISSUE TITLE
Batched method for getting the mask

### ISSUE THREAD

**Issue by trideeprath** [PAIR-code/saliency#10]
Is there a way to get the mask with batched images. 

Currently, the methods GetMask and GetSmoothedMask work with only one image as input.

**Comment by nalinimsingh** [comment]
We're interested in being able to generate masks for batched images as well -- would you be okay with incorporating this into the library? I'm happy to work on this, if so.

**Comment by trideeprath** [comment]
It would be great to have the batch feature included in the library if it increases the rate of generating the masks.

**Comment by nalinimsingh** [comment]
Right, makes sense. That comment was directed at the maintainers to make sure they’re alright with it — sorry for the confusion!

**Comment by trideeprath** [comment]
@nalinimsingh I can try if you can give a brief of the workaround. 
The current workaround I am using for faster saliency map is running it on multiple processes on multiple GPUs.

**Comment by nalinimsingh** [comment]
Sorry for the delayed response  -- I ended up writing my own implementation of GradCAM (elsewhere in a private codebase), and supported batching there.

**Comment by bwedin** [comment]
tl;dr We don't have plans to implement this on our end, but would be open to reviewing PRs that add this functionality.

Is this still relevant to your use cases? We implemented batching for individual masks (i.e. batching many steps of IG) in https://github.com/PAIR-code/saliency/pull/49, but don't have plans to implement batching for multiple images. This is primarily since the bottleneck for path-based methods is in calculating several steps at once, which is solved with the issue above. For non path-based methods (i.e. Gradients, Occlusion, Grad-CAM), there could create a new parameter that specifies the inputs are batched, and then returns batched masks since each method only calls the model once.
